### PR TITLE
select storage resources to attach to in storage service create

### PR DESCRIPTION
**Pull request by OrGur1987**
- storage service: ems refresh for service create and delete is general and not targeted, so that service-resource-attachments will be refreshed as well.

- cloud volume: create performs a general refresh if:
  - mode was advanced: so that the new service and its attachments to the selected resources will be refreshed as well.
  - count > 1: so that all the new volumes will be refreshed

- [ ] https://github.com/ManageIQ/manageiq-ui-classic/pull/8662

creating a storage with attached resources now enables to select it in volume create basic mode:
image

and the volumes will be created on one of its attached resources:
image

### CODE REVIEW

**Comment by OrGur1987** [comment]
@agrare :)

**Comment by OrGur1987** [comment]
@agrare 
https://github.com/ManageIQ/manageiq-ui-classic/pull/8662 approved by @jeffibm

File: app/models/manageiq/providers/autosde/storage_manager/cloud_volume.rb [hunk 1]
```diff
@@ -19,16 +19,25 @@ def self.raw_create_volume(ext_management_system, options = {})
 
     if options['mode'] == 'Basic'
       creation_hash[:service] = ext_management_system.storage_services.find(options["storage_service_id"]).ems_ref
+      if options["count"] == 0
+        target_class = :cloud_volumes
+        target_option = "new"
+      else
+        target_class = nil
+        target_option = "ems"
```
Comment (agrare) [inline app/models/manageiq/providers/autosde/storage_manager/cloud_volume.rb:27]: Follow-up, but we can do targeted refresh of multiple targets which will still be way more efficient than a full ems refresh.  Can you get the list of ems_refs of the new volumes?

File: app/models/manageiq/providers/autosde/storage_manager/storage_service.rb [hunk 2]
```diff
@@ -17,25 +18,25 @@ def self.raw_create_storage_service(ext_management_system, options = {})
...
     task_id = ext_management_system.autosde_client.ServiceApi.services_post(service_to_create).task_id
     options = {
-      :target_class   => :storage_services,
+      :target_class   => nil,
       :target_id      => nil,
       :ems_id         => ext_management_system.id,
       :native_task_id => task_id,
       :interval       => 10.seconds,
-      :target_option  => "new"
+      :target_option  => "ems"
```
Comment (agrare) [inline app/models/manageiq/providers/autosde/storage_manager/storage_service.rb:26]: Why this change?  Is the targeted refresh not working?  Seems like something we should fix not just do a full.

File: app/models/manageiq/providers/autosde/storage_manager/storage_service.rb [hunk 3]
```diff
@@ -17,25 +18,25 @@ def self.raw_create_storage_service(ext_management_system, options = {})
...
     options = {
-      :target_id      => id,
-      :target_class   => self.class.name,
+      :target_id      => nil,
+      :target_class   => nil,
       :ems_id         => ext_management_system.id,
       :native_task_id => task_id,
       :interval       => 20.seconds,
-      :target_option  => "existing"
+      :target_option  => "ems"
```
Comment (agrare) [inline app/models/manageiq/providers/autosde/storage_manager/storage_service.rb:39]: Same here, this looks unrelated to the rest of the changes and delete should be a targeted refresh.

Comment (OrGur1987) [reply app/models/manageiq/providers/autosde/storage_manager/storage_service.rb:39]: I've opened #220 for this particular change and will remove it from this PR

File: app/models/manageiq/providers/autosde/storage_manager/cloud_volume.rb (hunk 1 above)
Comment (OrGur1987) [reply app/models/manageiq/providers/autosde/storage_manager/cloud_volume.rb:27]: the advanced volume create leads to the backend first creating a new service, attaching it to the selected resources and creating the volume(s) on one of the resources. so we get the `ems_ref`s of the new volume(s), but not that of the new service and `storage_service_resource_attachments`.

if triggering a full refresh is too much of an overkill, we'll leave it as targeted for what we can get, and we'll just have to manually perform a full refresh until we manage to get `ems_ref`s for various types of objects in one job.

**Comment by OrGur1987** [comment]
@agrare following your comments, the current version has a targeted refresh on service create and delete and volume create basic, and a full one only for volume create advanced. if you prefer a targeted refresh on volume create advanced as well, I'll change it, and we'll have to manually refresh until we manage to get `ems_ref`s for various types of objects in one job.

**Comment by agrare** [comment]
@OrGur1987 no issue with handling the advanced case in a follow-up

**Review by agrare — APPROVED** [review]
